### PR TITLE
BasePool Docs (NatSpec)

### PR DESCRIPTION
**Pull request by EndymionJkb**
Extracted from recovery PR

### CODE REVIEW

File: pkg/pool-utils/contracts/BasePool.sol [hunk 1]
```diff
@@ -183,10 +206,21 @@ abstract contract BasePool is IBasePool, BasePoolAuthorization, BalancerPoolToke
...
+     */
     function pause() external authenticate {
         _setPaused(true);
     }
 
+    /**
+     * @notice Reverse a `pause` operation, and restore a pool to normal functionality.
+     * @dev This is a permissioned function that will only work on a paused pool within the Buffer Period set during
+     * pool factory deployment (see `TemporarilyPausable`). It does not automatically exit Recovery Mode. Note that
+     * any paused pools will automatically unpause after the Buffer Period expires.
```
Comment (TomAFrench) [inline pkg/pool-utils/contracts/BasePool.sol:222]:
```suggestion
     * pool factory deployment (see `TemporarilyPausable`). Note that any paused pools will automatically unpause after the Buffer Period expires.
```

File: pkg/pool-utils/contracts/LegacyBasePool.sol [hunk 2]
```diff
@@ -156,17 +169,34 @@ abstract contract LegacyBasePool is IBasePool, BasePoolAuthorization, BalancerPo
...
         bytes32 poolId = getPoolId();
         (, , , address assetManager) = getVault().getPoolTokenInfo(poolId, token);
 
         IAssetManager(assetManager).setConfig(poolId, poolConfig);
     }
 
+    /**
+     * @notice Pause the pool: an emergency action which disables all pool functions.
+     * @dev This is a permissioned function that will only work during the Pause Window set during pool factory
+     * deployment (see `TemporarilyPausable`). It also puts the pool in Recovery Mode, so that LPs can safely exit.
```
Comment (TomAFrench) [inline pkg/pool-utils/contracts/LegacyBasePool.sol:188]:
```suggestion
     * deployment (see `TemporarilyPausable`).
```

File: pkg/pool-utils/contracts/LegacyBasePool.sol [hunk 3]
```diff
@@ -156,17 +169,34 @@ abstract contract LegacyBasePool is IBasePool, BasePoolAuthorization, BalancerPo
...
+     */
     function pause() external authenticate {
         _setPaused(true);
     }
 
+    /**
+     * @notice Reverse a `pause` operation, and restore a pool to normal functionality.
+     * @dev This is a permissioned function that will only work on a paused pool within the Buffer Period set during
+     * pool factory deployment (see `TemporarilyPausable`). It does not automatically exit Recovery Mode. Note that
+     * any paused pools will automatically unpause after the Buffer Period expires.
```
Comment (TomAFrench) [inline pkg/pool-utils/contracts/LegacyBasePool.sol:198]:
```suggestion
     * pool factory deployment (see `TemporarilyPausable`). Note that any paused pools will automatically unpause after the Buffer Period expires.
```

File: pkg/pool-utils/contracts/BasePool.sol [hunk 4]
```diff
@@ -217,6 +251,10 @@ abstract contract BasePool is IBasePool, BasePoolAuthorization, BalancerPoolToke
         _;
     }
 
+    /**
+     * @notice Add liquidity to a pool (including for the first time, known as "initializing" the pool).
```
Comment (TomAFrench) [inline pkg/pool-utils/contracts/BasePool.sol:255]: I would make the `@notice` tag more explicit that this is a hook.

**Review by TomAFrench — CHANGES_REQUESTED** [review]
We've got a couple of things in here which are recovery mode specific so should be added in that PR.

File: pkg/pool-utils/contracts/BasePool.sol (hunk 4 above)
Comment (EndymionJkb) [reply pkg/pool-utils/contracts/BasePool.sol:255]: Noted on the exit as well.

File: pkg/pool-utils/contracts/BasePool.sol [hunk 5]
```diff
@@ -32,6 +32,7 @@ import "./BasePoolAuthorization.sol";
 // solhint-disable max-states-count
 
 /**
+ * @notice Reference implementation for the base layer of a Pool contract
```
Comment (TomAFrench) [inline pkg/pool-utils/contracts/BasePool.sol:35]:
```suggestion
 * @notice Reference implementation for the base layer of a Pool contract.
```

**Review by TomAFrench — APPROVED** [review]
Looks good, just a couple of nits

**Pull request merged**